Merge styles with Choice custom styles

ahnucl opened the pull request:
Allow selected and highlighted styles to work with Choice custom styles.

**What is the problem that this PR addresses?**
I needed the hightlighting functionality with custom styles no my Choices.

Closes https://github.com/tmbo/questionary/issues/151
...

**How did you solve it?**
I found that when the `title` param on Choice is a list it just `extend` the styles not checking if `selected` or `hightlighted` exists

...

**Checklist**
<!--- Put an `x` in all the boxes that apply. -->
<!-- Automated tests validate that your PR follows our guidelines. -->
<!-- If it doesn't pass, click on it to see details as to what your PR might be missing. -->

- [ ] I have read the [Contributor's Guide](https://questionary.readthedocs.io/en/stable/pages/contributors.html#steps-for-submitting-code).
- [ ] I will check that all automated PR checks pass before the PR gets reviewed.

ahnucl committed:
Merge styles with Choice custom styles

Allow selected and highlighted styles to work with Choice custom styles.

kiancross commented:
@ahnucl Please can you update this branch with latest changes from `master` and then I will review.

kiancross committed:
Clean up list-title style merge and add regression test

Rewrite the formatted-text branch as a generator expression (fixes
flake8 E203 and black line-length on the original lambdas) and add a
test covering highlighted/selected styling for Choice custom styles.

kiancross reviewed (approved):
Thanks for the PR!
